diff --git a/src/System/Taffybar/Information/SafeX11.hs b/src/System/Taffybar/Information/SafeX11.hs
index 5783602..280118d 100644
--- a/src/System/Taffybar/Information/SafeX11.hs
+++ b/src/System/Taffybar/Information/SafeX11.hs
@@ -114,23 +114,20 @@ requestQueue = unsafePerformIO newChan
 x11Thread :: ThreadId
 x11Thread = unsafePerformIO $ forkIO startHandlingX11Requests
 
-withErrorHandler :: XErrorHandler -> IO a -> IO a
-withErrorHandler new_handler action = do
-    handler <- mkXErrorHandler (\d e -> new_handler d e >> return 0)
-    original <- _xSetErrorHandler handler
-    res <- action
-    _ <- _xSetErrorHandler original
-    return res
-
-deriving instance Show ErrorEvent
+--withErrorHandler :: XErrorHandler -> IO a -> IO a
+--withErrorHandler new_handler action = do
+--    handler <- mkXErrorHandler (\d e -> new_handler d e >> return 0)
+--    original <- _xSetErrorHandler handler
+--    res <- action
+--    _ <- _xSetErrorHandler original
+--    return res
+--
+--deriving instance Show ErrorEvent
 
 startHandlingX11Requests :: IO ()
-startHandlingX11Requests =
-  withErrorHandler handleError handleX11Requests
-  where handleError _ xerrptr = do
-          ee <- getErrorEvent xerrptr
-          logHere WARNING $
-                  printf "Handling X11 error with error handler: %s" $ show ee
+startHandlingX11Requests = do
+  setDefaultErrorHandler
+  handleX11Requests
 
 handleX11Requests :: IO ()
 handleX11Requests = do
